__tests__/pet.test.js

```diff
@@ -0,0 +1,153 @@
+          const pet = new Pet("fido");
+
+          pet.fitness = 8;
+          pet.walk();
+
+          expect(pet.fitness).toEqual(10);
+        });
+        describe("feed", () => {
+          it("decreases hunger by 3 but never below 0", () => {
+            const pet = new Pet("fido");
+
+            pet.hunger = 2;
+            pet.feed();
+
+            expect(pet.hunger).toEqual(0);
+          });
+          describe("checkup", () => {
+            it("gives instruction if requires fitness or hungry", () => {
+              const pet = new Pet("fido");
+
+              pet.hunger = 5;
+              pet.fitness = 3;
+              pet.checkup();
+
+              expect(pet.fitness).toEqual("I am hungry AND I need a walk");
+              expect(pet.hunger).toEqual("I am hungry AND I need a walk");
+            });
+
+            it("gives instruction if pet needs a walk", () => {
+              const pet = new Pet("fido");
+
+              pet.fitness = 2;
+              pet.hunger = 4;
+              pet.checkup();
+              expect(pet.fitness).toEqual("I need a walk");
+            });
+            it("gives instruction if pet is hungry", () => {
+              const pet = new Pet("fido");
+
+              pet.fitness = 4;
+              pet.hunger = 5;
+              pet.checkup();
+              expect(pet.hunger).toEqual("I am hungry");
+            });
+            it("pet is feeling great", () => {
+              const pet = new Pet("fido");
+
+              pet.fitness = 4;
+              pet.hunger = 4;
+              pet.checkup();
+              expect(pet.hunger).toEqual("I feel great!");
+              expect(pet.fitness).toEqual("I feel great!");
+            });
+            describe("get isAlive", () => {
+              it("pets hunger, fitness and age returning true or false", () => {
+                const pet = new Pet("fido");
+
+                pet.hunger = 11;
+                pet.fitness = 0;
+                pet.age = 31;
+
+                expect(pet.isAlive).toEqual(false);
+              });
+              it("pets hunger, fitness and age returning true or false", () => {
+                const pet = new Pet("fido");
+
+                pet.hunger = 9;
+                pet.fitness = 2;
+                pet.age = 29;
+
+                expect(pet.isAlive).toEqual(true);
+              });
+              describe(" walk, feed, checkup", () => {
+                it("throws an error if the pet is not alive", () => {
+                  const pet = new Pet("Fido");
+
+                  pet.age = 30;
+                  pet.hunger = 11;
+                  pet.fitness = 0;
+                  
+                  expect(() => pet.walk()).toThrow("Your pet is no longer alive :(");
+                  expect(() => pet.feed()).toThrow("Your pet is no longer alive :(");
+                  expect(() => pet.checkup()).toThrow("Your pet is no longer alive :(");
+                });
```

Comment:
This should be broken up into three separate tests, as right now you need all three of these conditions to be met in order for the test to pass. See the suggestion on the `pet.js` file for a solution to this

Also, since you are testing the various methods (`walk`, `feed` and `checkUp`) here these test should be grouped with the other tests for those methods